src/Symfony/Component/Ldap/Ldap.php

```diff
@@ -0,0 +1,165 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Ldap;
+
+use Symfony\Component\Ldap\Connection\Connection;
+use Symfony\Component\Ldap\Connection\ConnectionInterface;
+use Symfony\Component\Ldap\Search\Query;
+
+/**
+ * @author Grégoire Pineau <lyrixx@lyrixx.info>
+ * @author Francis Besset <francis.besset@gmail.com>
+ * @author Charles Sarrazin <charles@sarraz.in>
+ */
+class Ldap implements LdapInterface
+{
+    private $connection;
+    private $charmaps;
+
+    /**
+     * Constructor.
+     *
+     * @param ConnectionInterface $connection A configured Ldap connection
+     */
+    public function __construct(ConnectionInterface $connection = null)
+    {
+        $this->connection = $connection ?: new Connection();
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function bind($dn = null, $password = null)
+    {
+        if (!$this->connection->isConnected()) {
```

Comment:
There is a double check: here and in `ConnectionInterface::bind`.

I'm not sure the interface should expose `ConnectionInterface::isConnected`.
IMHO, it's better to not expose the internal state, and "just try to bind" here.

it could be the responsibility of the connection to lazy-load an connect when you call bind.
It will make the code simpler.